src/Symfony/Component/Form/Extension/Core/Type/FileType.php

```diff
@@ -12,15 +12,40 @@
 namespace Symfony\Component\Form\Extension\Core\Type;
 
 use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\Options;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
 class FileType extends AbstractType
 {
     /**
      * {@inheritdoc}
      */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if ($options['multiple']) {
+            $builder->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $event) {
+                $form = $event->getForm();
+                $data = $event->getData();
+
+                // submitted data for an input file (not required) without choosing any file
+                if (array(null) === $data) {
```

Comment:
Not sure it needs a data transformer, as this is only about filtering out null values. Again; to enforce the `multiple="true"` effect server-side. It's an edge case.

This does not apply in any way to other types. Doing weird things with textypes, ie. changing  `name="texttype` to `name="texttype[weird]"`, will make it crash already (a good thing);

```
Expected argument of type "string", "array" given
```